reflex/ivars/object.py

```diff
@@ -581,30 +616,103 @@ def _cached_var_name(self) -> str:
 class ObjectMergeOperation(ObjectVar):
     """Operation to merge two objects."""
 
-    left: ObjectVar = dataclasses.field(default_factory=lambda: LiteralObjectVar({}))
-    right: ObjectVar = dataclasses.field(default_factory=lambda: LiteralObjectVar({}))
+    _lhs: ObjectVar = dataclasses.field(
+        default_factory=lambda: LiteralObjectVar.create({})
+    )
+    _rhs: ObjectVar = dataclasses.field(
+        default_factory=lambda: LiteralObjectVar.create({})
+    )
+
+    @cached_property
+    def _cached_var_name(self) -> str:
+        """The name of the operation.
+
+        Returns:
+            The name of the operation.
+        """
+        return f"Object.assign({self._lhs._var_name}, {self._rhs._var_name})"
```

Comment:
seems unsafe to mutate `self._lhs`, is there a reason to not use the object spread syntax?

```
return f"{{...{str(self._lhs)}, ...{str(self._rhs)}}}
```